### ISSUE TITLE
VSCode plugin shows "Failed to connect with build server, no functionality will work", cannot retrieve the reason why the connection failed

### ISSUE THREAD

**Issue by tribbloid** [scalameta/metals#2675]
When importing a project in scala 2.13. VSCode metal shows the following message:

2021.04.02 17:30:30 ERROR Failed to connect with build server, no functionality will work.
java.lang.RuntimeException: The server did not start, got FailedToOpenBspConnection
	at bloop.launcher.LauncherMain.failPromise$1(Launcher.scala:95)
	at bloop.launcher.LauncherMain.runLauncher(Launcher.scala:122)
	at scala.meta.internal.metals.BloopServers$$anon$1.run(BloopServers.scala:159)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**To Reproduce**

1. checkout gradle project https://github.com/tribbloid/shapesafe/tree/b4f7b0eccd96049c5cda55ee326c2e9583941d96 from github
2. import into VSCode with scala plugin

**Expected behavior**

The gradle project should be converted into bloop and compiled by a metal server instance

**Screenshots**

![Screenshot from 2021-04-02 17-36-56](https://user-images.githubusercontent.com/3441413/113456014-176b6580-93da-11eb-9e33-7b29741f9557.png)

**Installation:**

- Operating system: Ubuntu LTS 18.03
- Editor: Visual Studio Code
- Metals version: v1.10.0

**Additional context**

The scala project is originally written with gradle kts, compiled with gradlew 6.8.3

To further identify this issue I need to read the log which shows how the metal server executes and terminates, but I can't find any log for this.

**Search terms**

gradle, scala-2.13.5

**Comment by tribbloid** [comment]
Looks like the problem can be addressed by deleting .bloop directory, But one question remains: how to read the metals log to know the reason of the connection failure?

**Comment by ckipp01** [comment]
Thanks for the report @tribbloid. So looking at your trace, this is something we've been hitting a bit lately. The issue ends up being that Metals is unable to successfully create a BSP connection with Bloop, and therefore a lot of the expected functionality won't work.

> Looks like the problem can be addressed by deleting .bloop directory

I'm not so sure that this is what actually fixes it, but rather when it attempts to connect another time is succeeds. Am I correct in that this is working for you now? In the future if that happens you can also manually try to connect via `Metals: Connect to Build Server` via the command pallette.

> But one question remains: how to read the metals log to know the reason of the connection failure?

So you're actually seeing the logs in your Metals output. Those are the logs.

**Comment by Giggiux** [comment]
I solved the same issue, thanks to @tgodzik, by defining the proper `JAVA_HOME` for bloop in the `.bloop` folder, as defined in [Bloop's docs](https://scalacenter.github.io/bloop/docs/server-reference#custom-java-options).
In my case it was java 11, instead of java 15.

In my case `Metals: Connect to Build Server` via the command pallette didn't work because bloop was still starting with the wrong jdk.
I had to manually kill the process and then restart Visual Studio Code, to make Bloop use the newly set JDK.

Hope this can help!
